core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -395,7 +396,7 @@ public FormValidation doValidateProxy(
             GetMethod method = null;
             try {
                 method = new GetMethod(testUrl);
-                method.getParams().setParameter("http.socket.timeout", DEFAULT_CONNECT_TIMEOUT_MILLIS > 0 ? DEFAULT_CONNECT_TIMEOUT_MILLIS : new Integer(30 * 1000));
+                method.getParams().setParameter("http.socket.timeout", DEFAULT_CONNECT_TIMEOUT_MILLIS > 0 ? DEFAULT_CONNECT_TIMEOUT_MILLIS : TimeUnit.SECONDS.toMillis(30));
```

Comment:
This caused [JENKINS-57383](https://issues.jenkins-ci.org/browse/JENKINS-57383). Proxy form validation no longer works at all.